setup_wordpress.sh

```diff
@@ -4,3 +4,68 @@ set -e
+
+echo '---------- Update & Upgrade APT ----------'
+sudo apt-get update -y
+sudo apt-get upgrade -y
+
+echo '---------- Install Nginx Package and configure Firewall ----------'
+sudo apt-get install nginx -y
+sudo ufw allow 'Nginx Full'
+
+echo '---------- Install MYSQL Packages ----------'
+sudo apt-get install mysql-server -y
+
+echo '---------- Create WordPress Database & User ----------'
+mysql_pw='YourPassword123!'
+sudo debconf-set-selections <<< "mysql-server mysql-server/root_password password ${mysql_pw}"
+sudo debconf-set-selections <<< "mysql-server mysql-server/root_password_again password ${mysql_pw}"
+sudo mysql -u root -p$mysql_pw < /vagrant/wpDB.sql
+
+echo '---------- Install PHP Packages ----------'
+sudo apt install php7.4-cli php7.4-fpm php7.4-mysql php7.4-json php7.4-opcache php7.4-mbstring php7.4-xml php7.4-gd php7.4-curl -y
+
+echo '---------- Self-SSL Certificate ----------'
+sudo openssl req -x509 -newkey rsa:2048 -days 365 -nodes \
+	-subj "/C=AR/ST=BSAS/L=BSAS/O=WordPress/OU=WP/CN=192.168.50.2" \
+    -keyout /etc/ssl/private/nginx-selfsigned.key -out /etc/ssl/certs/nginx-selfsigned.crt
+sudo openssl dhparam -out /etc/nginx/dhparam.pem 2048
+
+echo "------------------ Create Configuration Snippet Pointing to the SSL Key and Certificate ------------------"
+sudo cp /vagrant/self-signed.conf /etc/nginx/snippets/
+sudo cp /vagrant/ssl-params.conf /etc/nginx/snippets/
+
+echo "------------------ WordPress Nginx Directory ------------------"
+WP_directory=/var/www/wordpress
+if [ ! -d "$WP_directory" ]; then
+    sudo mkdir ${WP_directory}
+fi
+
+echo "------------------ Nginx WordpressSite File Configuration && Symbolic link ------------------"
+sudo cp /vagrant/WPNginxConfig.conf /etc/nginx/sites-available/wordpress
+sudo ln -s /etc/nginx/sites-available/wordpress /etc/nginx/sites-enabled/
+sudo unlink /etc/nginx/sites-enabled/default
+sudo nginx -t
+sudo systemctl restart nginx
+
+echo "------------------ WordPress Download & Extract ------------------"
+cd /tmp
+sudo wget https://wordpress.org/latest.tar.gz
+sudo tar xvf latest.tar.gz
+
+echo "------------------ WordPress Privileges Setup && Wordpress PHP Configuration ------------------"
+sudo cp -a /tmp/wordpress/* ${WP_directory}
+sudo chown -R www-data:www-data ${WP_directory}
+sudo cp /vagrant/wp-config.php /var/www/wordpress/wp-config.php
+
+echo "------------------  WordPress CLI Installation ------------------"
+sudo curl -O https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar
+sudo chmod +x wp-cli.phar
+sudo mv wp-cli.phar /usr/local/bin/wp
+
+echo "------------------  WordPress User && Wordpress Theme ------------------"
+cd /var/www/wordpress
+sudo wp core install --url="192.168.50.2" --title="WordPress Setup" --admin_user=admin --admin_email=admin123@admin.com --admin_password="!2three456." --allow-root
+sudo wp theme install "twentynineteen" --activate --allow-root
+
+sudo apt-get install w3m -y
```

Comment:
where do you use `w3m`?

Reply:
Because w3m lets me check if the webpage is on, it lets navigate through the command line interface.

Reply:
but you can use your browser and connect to `192.168.50.2`

Reply:
Yes, you are true. I tried that and worked.